src/SprykerEco/Zed/Unzer/Business/Payment/Filter/UnzerEnabledPaymentMethodFilter.php

```diff
@@ -59,65 +60,130 @@ public function filterPaymentMethods(
         PaymentMethodsTransfer $paymentMethodsTransfer,
         QuoteTransfer $quoteTransfer
     ): PaymentMethodsTransfer {
-        if (!$this->hasUnzerPaymentMethods($paymentMethodsTransfer)) {
+        if (!$this->hasUnzerPaymentMethod($paymentMethodsTransfer)) {
             return $paymentMethodsTransfer;
         }
 
-        $filteredPaymentMethodTransfers = $this->hasMultipleMerchants($quoteTransfer)
-            ? $this->getMarketplaceFilteredPaymentMethods($paymentMethodsTransfer, $quoteTransfer)
-            : $this->getStandardFilteredPaymentMethods($paymentMethodsTransfer, $quoteTransfer);
+        if ($quoteTransfer->getUnzerCredentials() && $quoteTransfer->getUnzerCredentialsOrFail()->getType() === UnzerConstants::UNZER_CREDENTIALS_TYPE_STANDARD) {
+            return $this->getStandardUnzerPaymentMethods($paymentMethodsTransfer, $quoteTransfer);
+        }
+
+        if ($this->hasMultipleMerchants($quoteTransfer)) {
+            return $this->getMarketplaceUnzerPaymentMethods($paymentMethodsTransfer, $quoteTransfer);
+        }
 
-        return $paymentMethodsTransfer->setMethods($filteredPaymentMethodTransfers);
+        return $this->getMerchantMarketplaceUnzerPaymentMethods($paymentMethodsTransfer, $quoteTransfer);
     }
 
     /**
      * @param \Generated\Shared\Transfer\PaymentMethodsTransfer $paymentMethodsTransfer
      * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
      *
-     * @return \ArrayObject<array-key, \Generated\Shared\Transfer\PaymentMethodTransfer>
+     * @return \Generated\Shared\Transfer\PaymentMethodsTransfer
      */
-    protected function getStandardFilteredPaymentMethods(
+    protected function getStandardUnzerPaymentMethods(
         PaymentMethodsTransfer $paymentMethodsTransfer,
         QuoteTransfer $quoteTransfer
-    ): ArrayObject {
-        $unzerKeypairTransfer = $this->getUnzerKeypair(
-            $quoteTransfer->getStoreOrFail(),
-            [UnzerConstants::UNZER_CREDENTIALS_TYPE_STANDARD, UnzerConstants::UNZER_CREDENTIALS_TYPE_MARKETPLACE_MAIN_MERCHANT],
+    ): PaymentMethodsTransfer {
+        $standardUnzerKeypairTransfer = $this->getStandardUnzerKeypair($quoteTransfer);
+        $standardPaymentMethods = new ArrayObject(
+            array_filter((array)$paymentMethodsTransfer->getMethods(), function (PaymentMethodTransfer $paymentMethodTransfer) {
+                return !$this->isUnzerPaymentProvider($paymentMethodTransfer) || !$this->isMarketplaceUnzerPaymentMethod($paymentMethodTransfer);
```

Comment:
Also better to convert this to XOR comparison

Reply:
Fixed.